pkg/workflows/sdk/v2/testutils/runtime_test.go

```diff
@@ -0,0 +1,324 @@
+package testutils_test
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"google.golang.org/protobuf/proto"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/capabilities/v2/protoc/pkg/test_capabilities/actionandtrigger"
+	actionandtriggermock "github.com/smartcontractkit/chainlink-common/pkg/capabilities/v2/protoc/pkg/test_capabilities/actionandtrigger/action_and_triggermock"
+	"github.com/smartcontractkit/chainlink-common/pkg/capabilities/v2/protoc/pkg/test_capabilities/basicaction"
+	basicactionmock "github.com/smartcontractkit/chainlink-common/pkg/capabilities/v2/protoc/pkg/test_capabilities/basicaction/basic_actionmock"
+	"github.com/smartcontractkit/chainlink-common/pkg/capabilities/v2/protoc/pkg/test_capabilities/basictrigger"
+	basictriggermock "github.com/smartcontractkit/chainlink-common/pkg/capabilities/v2/protoc/pkg/test_capabilities/basictrigger/basic_triggermock"
+	"github.com/smartcontractkit/chainlink-common/pkg/capabilities/v2/protoc/pkg/test_capabilities/nodeaction"
+	nodeactionmock "github.com/smartcontractkit/chainlink-common/pkg/capabilities/v2/protoc/pkg/test_capabilities/nodeaction/node_actionmock"
+	"github.com/smartcontractkit/chainlink-common/pkg/workflows/sdk/v2"
+	"github.com/smartcontractkit/chainlink-common/pkg/workflows/sdk/v2/pb"
+	"github.com/smartcontractkit/chainlink-common/pkg/workflows/sdk/v2/testutils"
+)
+
+func TestRuntime_CallCapabilityIsAsync(t *testing.T) {
+	anyConfig := &basictrigger.Config{Name: "name", Number: 123}
+	anyTrigger := &basictrigger.Outputs{CoolOutput: "cool"}
+
+	trigger, err := basictriggermock.NewBasicCapability(t)
+	require.NoError(t, err)
+	trigger.Trigger = func(_ context.Context, config *basictrigger.Config) (*basictrigger.Outputs, error) {
+		assert.True(t, proto.Equal(anyConfig, config))
+		return anyTrigger, nil
+	}
+
+	ch := make(chan struct{}, 1)
+	anyResult1 := "ok1"
+	action1, err := basicactionmock.NewBasicActionCapability(t)
+	require.NoError(t, err)
+	action1.PerformAction = func(_ context.Context, input *basicaction.Inputs) (*basicaction.Outputs, error) {
+		<-ch
+		return &basicaction.Outputs{AdaptedThing: anyResult1}, nil
+	}
+
+	anyResult2 := "ok2"
+	action2, err := actionandtriggermock.NewBasicCapability(t)
+	action2.Action = func(ctx context.Context, input *actionandtrigger.Input) (*actionandtrigger.Output, error) {
```

Comment:
I feel like we should leverage Mockery here instead of building it ourselves...

Reply:
It does more when you call new. I also like that a single tool is used for everything instead of two. Let's leave this decision for the Go Guild to discuss more. I showed them this before, and I didn't get any objections.